unit_tests/test_pairwise_alltoallv.cpp

```diff
@@ -13,47 +14,75 @@ int main(int argc, char *argv[]) {
   int comm_rank;
   MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank);
 
-  // Make an array of counts for other processors
-  std::vector<int> sendcounts(comm_size);
-  for(int i=0; i<comm_size; i+=1) {
-    sendcounts[i] = 100+comm_rank*50;
-  }
+  // Set up repeatable RNG with different seed on each rank
+  std::mt19937 rng;
+  rng.seed(comm_rank);
+
+  const int nr_reps = 1000;
+  for(int rep_nr=0; rep_nr<nr_reps; rep_nr+=1) {
 
-  // Allocate the send buffer
-  int nr_local_elements = 0;
-  for(auto n : sendcounts)
-    nr_local_elements += n;
-  std::vector<int> sendbuf(nr_local_elements);
-
-  // Populate the send buffer with index of the source and destination ranks
-  int offset = 0;
-  for(int dest=0; dest<comm_size; dest+=1) {
-    for(int i=0; i<sendcounts[dest]; i+=1) {
-      sendbuf[offset] = comm_rank + 1000*dest;
-      offset += 1;
+    // Determine maximum message size for small chunks test (all ranks need to agree on this)
+    std::uniform_int_distribution<> send_size_dist(4, 128);
```

Comment:
Any reason as to why the ranges are different to `std::uniform_int_distribution<> send_size_dist(1, 100);` in `test_myalltoall.cpp`?

Reply:
I should rename those variables because they mean different things. In pairwise_alltoall the chunk size is in bytes. In MyAlltoall it's in array elements. The minimums are different because if the maximum send size in bytes is less than one array element the pairwise alltoall function can't work.